app/javascript/components/Search.js

```diff
@@ -1,107 +1,114 @@
-        this.state = {
-            query: "",
-            searchResult: {pages:{}, docs:[], facets:[]},
-            currentPage: 1,
-            appliedFacets: [],
-            perPage: 10
-        };
-    }
+  constructor(props) {
+    super(props);
+    this.state = {
+      query: '',
+      searchResult: { pages: {}, docs: [], facets: [] },
+      currentPage: 1,
+      appliedFacets: [],
+      perPage: 10,
+      filterURL: props.baseUrl + '?f[collection_ssim][]=' + props.collection
+    };
+  }
 
-    handleQueryChange = event => {
-        this.setState({query: event.target.value, currentPage: 1});
-    }
+  handleQueryChange = event => {
+    this.setState({ query: event.target.value, currentPage: 1 });
+  };
 
-    componentDidMount() {
-        this.retrieveResults();
-    }
+  componentDidMount() {
+    this.retrieveResults();
+  }
 
-    componentDidUpdate(prevProps, prevState) {
-        if (prevState.query != this.state.query || prevState.currentPage != this.state.currentPage || prevState.appliedFacets != this.state.appliedFacets) {
-            this.retrieveResults();
-        }
+  componentDidUpdate(prevProps, prevState) {
+    if (
+      prevState.query != this.state.query ||
+      prevState.currentPage != this.state.currentPage ||
+      prevState.appliedFacets != this.state.appliedFacets
+    ) {
+      this.retrieveResults();
     }
+  }
 
-    retrieveResults() {
-        let component = this;
-        let facetFilters = "";
-        this.state.appliedFacets.forEach((facet) => { facetFilters = facetFilters + "&f[" + facet.facetField + "][]=" + facet.facetValue });
-        if (this.props.collection) {
-            facetFilters = facetFilters + "&f[collection_ssim][]=" + this.props.collection;
-        }
-        let url = this.props.baseUrl + "/catalog.json?per_page=" + this.state.perPage + "&q=" + this.state.query + "&page=" + this.state.currentPage + facetFilters;
-        console.log("Performing search: " + url);
-        Axios({url: url})
-            .then(function(response){
-                console.log(response);
-                component.setState({searchResult: response.data.response})
-            });
+  retrieveResults() {
+    let component = this;
+    let facetFilters = '';
+    this.state.appliedFacets.forEach(facet => {
+      facetFilters =
+        facetFilters + '&f[' + facet.facetField + '][]=' + facet.facetValue;
+    });
+    if (this.props.collection) {
+      facetFilters =
+        facetFilters + '&f[collection_ssim][]=' + this.props.collection;
     }
+    let url =
+      this.props.baseUrl +
+      '/catalog.json?per_page=' +
+      this.state.perPage +
+      '&q=' +
+      this.state.query +
+      '&page=' +
+      this.state.currentPage +
+      facetFilters;
+    console.log('Performing search: ' + url);
+    Axios({ url: url }).then(function(response) {
+      console.log(response);
+      component.setState({ searchResult: response.data.response });
+    });
+  }
```

Comment:
Seems all the string concatenation could be more readable using String template literals?   https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals